main/settings/src/main/scala/sbt/Reference.scala

```diff
@@ -78,7 +78,7 @@ object Reference {
       case LocalRootProject    => "{<this>}<root>"
       case LocalProject(id)    => "{<this>}" + id
       case RootProject(uri)    => "{" + uri + " }<root>"
-      case ProjectRef(uri, id) => "{" + uri + "}" + id
+      case ProjectRef(uri, id) => s"""ProjectRef(uri("$uri"),"$id")"""
```

Comment:
Last comment sounds harsh. I'm gonna reword in a nicer tone:

1. Why a new function?
2. Which of my previous arguments don't you agree with?
3. Why do you still this is a good idea (when most of the builds will **never** use project references specifying the URI)?

In my opinion, I'm proposing something reasonable, that addresses your main concerns and that is objectively better: more concise, shorter, clearer and natural to end users.